internal/filter/filter.go

```diff
@@ -18,18 +18,60 @@
 package filter
 
 import (
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/sirupsen/logrus"
 	"open-match.dev/open-match/pkg/pb"
 )
 
 var emptySearchFields = &pb.SearchFields{}
 
-// InPool returns whether the ticket meets all the criteria of the pool.
-func InPool(ticket *pb.Ticket, pool *pb.Pool) bool {
+var (
+	logger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "filter",
+	})
+)
+
+// Criteria specifies all the criteria that the Ticket needs to meet to
+// be selected. This is used instead of the Pool directly to avoid any
+// criteria validation logic in the filter package.
+type Criteria struct {
+	DoubleRangeFilters  []*pb.DoubleRangeFilter
+	StringEqualsFilters []*pb.StringEqualsFilter
+	TagPresentFilters   []*pb.TagPresentFilter
+	CreatedBefore       time.Time
+	CreatedAfter        time.Time
+}
+
+// MeetsCriteria returns whether the ticket meets all the criteria of the pool.
+func MeetsCriteria(ticket *pb.Ticket, fc *Criteria) bool {
 	s := ticket.GetSearchFields()
 	if s == nil {
 		s = emptySearchFields
 	}
-	for _, f := range pool.GetDoubleRangeFilters() {
+
+	// CreateTime is only populated by Open Match and hence expected to be valid.
+	if ct, err := ptypes.Timestamp(ticket.CreateTime); err == nil {
+		if !fc.CreatedAfter.IsZero() {
+			if !ct.After(fc.CreatedAfter) {
+				return false
+			}
+		}
+
+		if !fc.CreatedBefore.IsZero() {
+			if !ct.Before(fc.CreatedBefore) {
+				return false
+			}
+		}
+	} else {
+		logger.WithFields(logrus.Fields{
```

Comment:
if we keep this logging, the ticket id should be logged.

Reply:
Done.